partcad/src/partcad/test/cam.py

```
@@ -0,0 +1,91 @@
+            manufacturing_data = AssemblyConfiguration.get_manufacturing_data(assembly)
+            if not manufacturing_data.method:
+                pc_logging.error("%s:%s can't be assembled" % (assembly.project_name, assembly.name))
+                # TODO(clairbee): Verify that at least one provider is available
+                return
+
+            # Now test if all of the parts for assembly are manufacturable by themselves
+            bom = await assembly.get_bom()
+            for part_name in bom:
+                part = ctx.get_part(part_name)
+                if part is None:
+                    pc_logging.error("Part %s:%s is missing" % (assembly.project_name, part_name))
```

Ruff (0.8.2)

Use format specifiers instead of percent format

Replace with format specifiers

(UP031)